iis-wf/iis-wf-transformers/src/main/resources/eu/dnetlib/iis/wf/transformers/metadataextraction/skip_extracted_without_meta/oozie_app/lib/scripts/transformer.pig

```diff
@@ -0,0 +1,24 @@
+define avro_load_document_content
+org.apache.pig.piggybank.storage.avro.AvroStorage(
+'schema', '$schema_document_content');
+
+define avro_load_document_meta
+org.apache.pig.piggybank.storage.avro.AvroStorage(
+'schema', '$schema_document_meta');
+
+define avro_store_document_content
+org.apache.pig.piggybank.storage.avro.AvroStorage(
+'index', '0',
+'schema', '$schema_document_content');
+
+documentContent = load '$input_document_content' using avro_load_document_content;
+documentMeta = load '$input_document_meta' using avro_load_document_meta;
+
+cachedDocumentId = foreach documentMeta generate id;
+cachedDocumentIdDistinct = distinct cachedDocumentId;
+
+joinedDocumentContent = join documentContent by id left, cachedDocumentIdDistinct by id;
+joinedFilteredDocumentContent = filter joinedDocumentContent by cachedDocumentIdDistinct::id is null;
```

Comment:
In fact we do need an outer join here because while joining tables A and B we are interested in subset of records appearing only in A, not in B (translating into this particular case: we are looking for contents given as input which were NOT stored in cache so they could be processed by `metadataextraction` algorithm).

The filter condition is `cachedDocumentIdDistinct::id is null` not `cachedDocumentIdDistinct::id is not null` what could be replaced by inner join.